recipes/pcl/all/conanfile.py

```diff
@@ -0,0 +1,224 @@
+        "with_apps": False,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    # in case the project requires C++14/17/20/... the minimum compiler version should be listed
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("boost/1.82.0")
+        self.requires("eigen/3.3.9")
+        self.requires("flann/1.9.2")
+        if self.options.with_png:
+            self.requires("libpng/1.6.39")
+        if self.options.with_qhull:
+            self.requires("qhull/8.0.1")
+        if self.options.with_apps:
+            self.requires("qt/6.5.0")
+        if self.options.with_libusb:
+            self.requires("libusb/1.0.26")
+
+    def validate(self):
+        # validate the minimum cpp standard supported. For C++ projects only
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+        # in case it does not work in another configuration, it should validated here too
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["PCL_SHARED_LIBS"] = self.options.shared
+        tc.variables["WITH_OPENMP"] = self.options.with_openmp
+        tc.variables["WITH_LIBUSB"] = self.options.with_libusb
+        tc.variables["WITH_PNG"] = self.options.with_png
+        tc.variables["WITH_QHULL"] = self.options.with_qhull
+        tc.variables["WITH_CUDA"] = self.options.with_cuda
+        tc.variables["WITH_PCAP"] = self.options.with_pcap
+        tc.variables["WITH_OPENGL"] = self.options.with_opengl
+        tc.variables["BUILD_TOOLS"] = self.options.with_tools
+        tc.variables["BUILD_APPS"] = self.options.with_apps
+        tc.cache_variables["WITH_QT"] = self.options.with_apps
+        tc.cache_variables["WITH_VTK"] = False
```

Comment:
Maybe add with_vtk option, but display a self.output.warning() mentioning that it needs to be provided by the system?